src/routes/safe/store/actions/createTransaction.ts

```diff
@@ -12,10 +14,52 @@ import { ZERO_ADDRESS } from 'src/logic/wallets/ethAddresses'
 import { EMPTY_DATA } from 'src/logic/wallets/ethTransactions'
 import { providerSelector } from 'src/logic/wallets/store/selectors'
 import { SAFELIST_ADDRESS } from 'src/routes/routes'
-import fetchTransactions from 'src/routes/safe/store/actions/fetchTransactions'
+import { addOrUpdateCancellationTransactions } from 'src/routes/safe/store/actions/transactions/addOrUpdateCancellationTransactions'
+import { addOrUpdateTransactions } from 'src/routes/safe/store/actions/transactions/addOrUpdateTransactions'
+import { removeCancellationTransaction } from 'src/routes/safe/store/actions/transactions/removeCancellationTransaction'
+import { removeTransaction } from 'src/routes/safe/store/actions/transactions/removeTransaction'
+import { mockTransaction } from 'src/routes/safe/store/actions/transactions/utils/transactionHelpers'
 import { getLastTx, getNewTxNonce, shouldExecuteTransaction } from 'src/routes/safe/store/actions/utils'
-
 import { getErrorMessage } from 'src/test/utils/ethereumErrors'
+import { makeConfirmation } from '../models/confirmation'
+import fetchTransactions from './transactions/fetchTransactions'
+import { safeTransactionsSelector } from 'src/routes/safe/store/selectors'
+
+export const removeTxFromStore = (tx, safeAddress, dispatch, state) => {
+  if (tx.isCancellationTx) {
+    const newTxStatus = 'awaiting_confirmations'
+    const transactions = safeTransactionsSelector(state)
+    const txsToUpdate = transactions
+      .filter((transaction) => Number(transaction.nonce) === Number(tx.nonce))
+      .withMutations((list) => list.map((tx) => tx.set('status', newTxStatus)))
+
+    batch(() => {
+      dispatch(addOrUpdateTransactions({ safeAddress, transactions: txsToUpdate }))
+      dispatch(removeCancellationTransaction({ safeAddress, transaction: tx }))
+    })
```

Comment:
well... here is the complicated thing about this mocked transactions. I'm looking with a real interest in implementing a new state in the store for pending transactions.

To clarify,

- `removeTxFromStore` is only used for those cases when we create a new transaction (this can happen if we can't try an off-chain signing or if there's only one user in the wallet), and the transaction submits fails. Thus, we remove it from the list (this is valid for `transactions` and `cancellationTransactions`).
  - With the later I realize that the `const newTxStatus = 'awaiting_confirmations'` is incorrect, it should have been `awaiting_your_confirmation`
- `storeTx` is used in all other cases. That is when a new transaction is created, when we need to rollback a `processTransaction` (an intention to execute a tx or "vote" for a tx when a it can be signed off-chain), and when we need to execute a tx.

This resulted in a very convoluted logic, and really prone to error.